doc/source/whatsnew/v2.2.0.rst

```diff
@@ -338,7 +338,9 @@ Groupby/resample/rolling
 Reshaping
 ^^^^^^^^^
 - Bug in :func:`concat` ignoring ``sort`` parameter when passed :class:`DatetimeIndex` indexes (:issue:`54769`)
+- Bug in :func:`merge` reordering index levels when merging on a subset of levels (:issue:`34133`)
```

Comment:
Based on the failing tests, I think we should put this in the notable bug fixes

Reply:
Moved to notable section